contracts/core/tokens/rebalancing-libraries/StandardProposeLibrary.sol

```diff
@@ -99,6 +105,17 @@ library StandardProposeLibrary {
             "RebalancingSetToken.propose: Invalid or disabled proposed SetToken address"
         );
 
+        // Check proposed components on whitelist. This is to ensure managers are unable to add contract addresses
+        // to a propose that prohibit the set from carrying out an auction i.e. a token that only the manager possesses
+        IWhiteList rebalanceComponentWhitelist = IRebalancingSetFactory(_factory).rebalanceComponentWhitelist();
```

Comment:
If we do not believe that the factory will change (which it shouldn't), we could `cache` the rebalanceComponentWhitelist in a state variable

Reply:
+1